src/snowflake/snowpark/_internal/utils.py

```diff
@@ -868,6 +869,10 @@ def func_call_wrapper(*args, **kwargs):  # pragma: no cover
         if "_emit_ast" in func.__code__.co_varnames and "_emit_ast" not in kwargs:
             # No arguments, or single argument with function.
             if len(args) == 0 or (len(args) == 1 and isinstance(args[0], Callable)):
+                # this can be refactored to
+                #  - extract session
+                #  - check isinstance(session, snowflake.snowpark.Session)
+                #  - update kwargs["_emit_ast"] = value
```

Comment:
I noticed we are not checking `isinstance(session, Session)` so this would make it more consistent